src/mds/MDSDaemon.cc

```diff
@@ -823,18 +823,19 @@ int MDSDaemon::_handle_command(
     cpu_profiler_handle_command(argvec, ds);
   } else {
     // Give MDSRank a shot at the command
-    if (mds_rank) {
+    if (!mds_rank) {
+      ss << "MDS not active";
+      r = -EINVAL;
+    }
+    else {
       bool handled = mds_rank->handle_command(cmdmap, m, &r, &ds, &ss,
 					      need_reply);
-      if (handled) {
-        goto out;
+      if (!handled) {
+        // MDSDaemon doesn't know this command
+        ss << "unrecognized command! " << prefix;
```

Comment:
I can't seem to see this error message using e.g.:

````
$ bin/ceph tell mds.c foo
no valid command found; 10 closest matches:
session evict {<filters> [<filters>...]}
client evict {<filters> [<filters>...]}
session ls {<filters> [<filters>...]}
client ls {<filters> [<filters>...]}
session kill <int>
cpu_profiler status|flush
exit
respawn
injectargs <injected_args> [<injected_args>...]
config set <key> <value>
Error EINVAL: invalid command
````

My guess is that the ceph.in program is doing some pre-validation.

Reply:
@batrick 
In this case [this](https://github.com/ceph/ceph/blob/38f051c22af1def4a06427876ee2e5000046fd03/src/ceph.in#L534) is printing.

Yes, you are right. I didn't came across a situation that prints `unrecognized command!` in my testing too. But as it was already there in the previous logic, I tried to retain it. 

However, I can fix it. Please suggest.